File: docs/examples/record_api_rs/src/list.rs
use trailbase_client::{Client, CompareOp, Filter, ListArguments, ListResponse, Pagination};

pub async fn list(client: &Client) -> anyhow::Result<ListResponse<serde_json::Value>> {
  Ok(
    client
      .records("movies")
      .list(
        ListArguments::new()
          .with_pagination(Pagination::new().with_limit(3))
          .with_order(["rank"])
          .with_filters([
            // Multiple filters on same column: watch_time between 90 and 120 minutes
            Filter::new("watch_time", CompareOp::GreaterThanOrEqual, "90"),
            Filter::new("watch_time", CompareOp::LessThan, "120"),
            // Date range: movies released between 2020 and 2023
            Filter::new("release_date", CompareOp::GreaterThanOrEqual, "2020-01-01"),
            Filter::new("release_date", CompareOp::LessThanOrEqual, "2023-12-31"),
          ]),
      )
      .await?,
  )
}

File: docs/examples/record_api_py/record_api_py/list.py
from trailbase import Client, Filter, CompareOp, JSON_OBJECT


def list_movies(client: Client) -> list[JSON_OBJECT]:
    response = client.records("movies").list(
        limit=3,
        order=["rank"],
        filters=[
            # Multiple filters on same column: watch_time between 90 and 120 minutes
            Filter(column="watch_time", value="90", op=CompareOp.GREATER_THAN_OR_EQUAL),
            Filter(column="watch_time", value="120", op=CompareOp.LESS_THAN),
            # Date range: movies released between 2020 and 2023
            Filter(column="release_date", value="2020-01-01", op=CompareOp.GREATER_THAN_OR_EQUAL),
            Filter(column="release_date", value="2023-12-31", op=CompareOp.LESS_THAN_OR_EQUAL),
        ],
    )

    return response.records
